base_layer/wallet/src/transaction_service/storage/sqlite_db.rs

```diff
@@ -2567,77 +2605,86 @@ mod test {
         let db_path = format!("{}{}", db_folder, db_name);
 
         embed_migrations!("./migrations");
-        let conn = SqliteConnection::establish(&db_path).unwrap_or_else(|_| panic!("Error connecting to {}", db_path));
-
-        embedded_migrations::run_with_output(&conn, &mut std::io::stdout()).expect("Migration failed");
-
-        let inbound_tx = InboundTransaction {
-            tx_id: 1,
-            source_public_key: PublicKey::from_secret_key(&PrivateKey::random(&mut OsRng)),
-            amount: MicroTari::from(100),
-            receiver_protocol: ReceiverTransactionProtocol::new_placeholder(),
-            status: TransactionStatus::Pending,
-            message: "Yo!".to_string(),
-            timestamp: Utc::now().naive_utc(),
-            cancelled: false,
-            direct_send_success: false,
-            send_count: 0,
-            last_send_timestamp: None,
-        };
-        let inbound_tx_sql = InboundTransactionSql::try_from(inbound_tx).unwrap();
-        inbound_tx_sql.commit(&conn).unwrap();
+        let mut pool = SqliteConnectionPool::new(db_path.clone(), 1, true, true, Duration::from_secs(60));
+        pool.create_pool()
+            .unwrap_or_else(|_| panic!("Error connecting to {}", db_path));
+        // Note: For this test the connection pool is setup with a pool size of one; the pooled connection must go out
```

Comment:
Nit: this is fine, just a note that you could explicitly drop() the pool rather than using a {} scope which I think is a but more explicit and easy to use.